usaspending_api/search/v2/views/search.py

```diff
@@ -403,3 +407,115 @@ def post(self, request):
         response['results'] = results
 
         return Response(response)
+
+
+class SpendingByAwardVisualizationViewSet(APIView):
+
+    @total_ordering
+    class MinType(object):
+        def __le__(self, other):
+            return True
+
+        def __eq__(self, other):
+            return (self is other)
+    Min = MinType()
+
+    def post(self, request):
+        """Return all budget function/subfunction titles matching the provided search text"""
+        json_request = request.data
+        fields = json_request.get('fields', None)
+        filters = json_request.get('filters', None)
+        order = json_request.get('order', "asc")
+        limit = json_request.get('limit', 10)
+        page = json_request.get('page', 1)
+
+        if fields is None:
+            raise InvalidParameterException('Missing one or more required request parameters: fields')
+        if filters is None:
+            raise InvalidParameterException('Missing one or more required request parameters: filters')
+        if "award_type_codes" not in filters:
+            raise InvalidParameterException('Missing one or more required request parameters: filters["award_type_codes"]')
+        if order not in ["asc", "desc"]:
+            raise InvalidParameterException('Invalid value for order: {}'.format(order))
+
+        sort = json_request.get('sort', fields[0])
```

Comment:
Setting the default to `fields[0]` could result in an exception if `fields` is `None`. We may want to test & handle that scenario accordingly.

Reply:
I moved this section of code after checking for if `fields` is `None` at line 432:

```
        if fields is None:
            raise InvalidParameterException('Missing one or more required request parameters: fields')
```